simulator/dummy_perception_publisher/src/signed_distance_function.cpp

```diff
@@ -47,18 +50,23 @@ double AbstractSignedDistanceFunction::getSphereTracingDist(
 
 double BoxSDF::operator()(double x, double y) const
 {
-  const auto vec_global = tf2::Vector3(x, y, 0.0);
-  const auto vec_local = tf_global_to_local_.inverse()(vec_global);
+  const auto && vec_global = tf2::Vector3(x, y, 0.0);
+  const auto vec_local = tf_local_to_global_(vec_global);
 
   // As for signed distance field for a box, please refere:
   // https://www.iquilezles.org/www/articles/distfunctions/distfunctions.htm
   const auto sd_val_x = std::abs(vec_local.getX()) - 0.5 * length_;
   const auto sd_val_y = std::abs(vec_local.getY()) - 0.5 * width_;
   const auto positive_dist_x = std::max(sd_val_x, 0.0);
   const auto positive_dist_y = std::max(sd_val_y, 0.0);
-  const auto positive_dist = std::hypot(positive_dist_x, positive_dist_y);
+
+  const auto positive_dist =
+    sqrt(positive_dist_x * positive_dist_x + positive_dist_y * positive_dist_y);
+  if (positive_dist > 0.0) {
```

Comment:
I think this is always true

Reply:
You are right, in this application of point cloud generation the positive dist is always > 0.0. However, because the class name is SDF(signed distance field), I think considering negative distance is better for understanding for peple use laster. (Also, sphere tracing is usually explained with Signed Distance Field).